components/CardList/CardList.test.js

```diff
@@ -0,0 +1,88 @@
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import CardList from "./CardList";
+
+describe("CardList component", () => {
+  const medications = [
+    {
+      id: 1,
+      name: "Acetaminophen",
+      summary:
+        "a pain reliever commonly found in over-the-counter medications such as Tylenol.",
+      isBookmarked: false,
+    },
+    {
+      id: 2,
+      name: "Ibuprofen",
+      summary:
+        "a nonsteroidal anti-inflammatory drug (NSAID) that reduces inflammation and pain, found in products such as Advil or Motrin.",
+      isBookmarked: true,
+    },
+    {
+      id: 3,
+      name: "Aspirin",
+      summary:
+        "medication that belongs to the class of drugs called nonsteroidal anti-inflammatory drugs (NSAIDs). It works by inhibiting the production of prostaglandins, which are chemicals that cause pain, inflammation, and fever in the body.",
+      isBookmarked: false,
+    },
+  ];
+  const mockToggleBookmark = jest.fn();
+  test("search for medications by name or summary", () => {
+    render(<CardList medications={medications} />);
+    const searchInput = screen.getByPlaceholderText("Search...");
```

Comment:
No changes requested. Just a comment. 

I think the fact that we are grabbing an element by `getByPlaceholderText` rather than `getByLabelText` tells us that we are missing something in terms of accessibility (placeholder is not as accessible). 

Here is a nice [guide to how you can make an accessible search bar](https://jdbrewerhofmann.medium.com/how-to-create-an-accessible-search-bar-without-a-label-d686181c2556).